ethcore/src/types/filter.rs

```diff
@@ -59,7 +65,8 @@ impl Clone for Filter {
 			from_block: self.from_block.clone(),
 			to_block: self.to_block.clone(),
 			address: self.address.clone(),
-			topics: topics[..].to_vec()
+			topics: topics[..].to_vec(),
```

Comment:
Hmm, it can contain up to 4 entries - but that's true, we could substitute this with array of 4 `Option`. I will prepare a separate PR.